### DISCUSSION TITLE
Insert Blob value from parameter in Python

### DISCUSSION THREAD

**Discussion by DiptenduIDEAS** [duckdb/duckdb#11680] [Q&A]
### What happens?

I am trying to save the contents of a CSV file into a Blob field from a Python program (using Streamlit for UI part).

I have defined a function to perform the insertion

```
def insert_dataset(fileName, fileBytes):
    if 'connection' in st.session_state:
        connection = st.session_state['connection'] 
    #return connection.execute("INSERT INTO datasets VALUES (nextval('seq_dataset_id'), ?, get_current_timestamp()), ?::blob" , [fileName, fileBytes])
    return connection.execute(f"INSERT INTO datasets VALUES (nextval('seq_dataset_id'), ?, get_current_timestamp()), {fileBytes}::blob" , [fileName])
```

I have tried 2 approaches (using ? placeholder as well as f{String} but both are giving me errors.

```
fileName = uploaded_file.name
fileBytes = uploaded_file.getvalue()
result = db.insert_dataset(fileName, fileBytes)
```

`duckdb.duckdb.ParserException: Parser Error: syntax error at or near "b'Row ID,Order Priority,Discount,Unit Price,Shipping Cost,Customer ID,Customer Name,Ship Mode,Customer Segment,Product Category,Product Sub-Category,Product Container,Product Name,Product Base Margin,Region,State or Province,City,Postal Code,Order Date,Ship Date,Profit,Quantity ordered new,Sales,Order ID\r\n18606,Not Specified,0.01,2.88,0.5,2,Janice Fletcher,Regular Air,Corporate,Office Supplies,Labels,Small Box,Avery 49,0.36,Central,Illinois,Addison,60101,28/05/2012,30/05/2012,1.3`

What is the best way to do this?

### To Reproduce

Run the code as given above

### OS:

Windows

### DuckDB Version:

duckdb==0.10.1

### DuckDB Client:

Python

### Full Name:

Diptendu Dutta

### Affiliation:

IDEAS TIH

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Tishj** [comment]
https://duckdb.org/docs/guides/import/read_file
